CMakeLists.txt

```diff
@@ -294,6 +295,42 @@ else()
     message( STATUS "${red}No HIP/ROCm support: gpu_backend = ${gpu_backend}${plain}" )
 endif()
 
+#-------------------------------------------------------------------------------
+# OneMKL/SYCL support.
+message( "" )
+set( blaspp_use_onemkl false )  # output in blasppConfig.cmake.in
+# if cuda or hip were found gpu_backend was set appropriately
+if (gpu_backend MATCHES "^(onemkl|auto)$")
+
+    message( STATUS "${bold}Looking for OneMKL${not_bold} (gpu_backend = ${gpu_backend})" )
+    if (gpu_backend STREQUAL "onemkl")
+        find_package( MKL CONFIG REQUIRED HINTS "$ENV{MKL_ROOT}")
+    else()
+        find_package( MKL CONFIG QUIET HINTS "$ENV{MKL_ROOT}")
+    endif()
+    # message(STATUS "Available targets: ${MKL_IMPORTED_TARGETS}")
+
+    if (MKL_FOUND)
+        set( gpu_backend "onemkl" )
+        set( blaspp_defs_onemkl_ "-DBLAS_HAVE_ONEMKL;-DBLAS_FORTRAN_ADD_" )
+        set( blaspp_use_onemkl true )
+
+        # Uncomment to use CMake FindBLAS using BLA_VENDOR
+        # if (NOT BLA_VENDOR)
+        #    set( BLA_VENDOR Intel10_64lp )
+        # endif()
+
+        target_compile_options( blaspp PUBLIC -std=c++17 -fsycl )
```

Comment:
There's already 
```
target_compile_features( blaspp PUBLIC cxx_std_17 )
```
below, which is generally the preferred CMake mechanism, rather than explicitly setting flags that the compiler may or may not support. I don't know for `-fsycl` if there's any way to set it other than you did here.

Reply:
Yes, that caused problems with the Fortran files. 
Updated to:
`        target_compile_options( blaspp PUBLIC -std=c++17 -fsycl )`

Reply:
I realized that this causes problems for Fortran.
Fixed in 7d953f3

Reply:
I removed the -std=c++17.  This was also causing problems for Fortran files.
`        target_compile_options( blaspp PUBLIC -fsycl )
`